src/main/scala/strawman/collection/immutable/LazyList.scala

```diff
@@ -172,12 +172,13 @@ sealed abstract class LazyList[+A]
       else prefix.lazyAppendAll(nonEmptyPrefix.tail.flatMap(f))
     }
 
-  override final def zip[B](xs: collection.Iterable[B]): LazyList[(A, B)] =
+  override final def zip[B](xs: collection.Iterable[B]): LazyList[(A, B)] = zipWith(xs)((_, _))
```

Comment:
Why do we need to override it here?

Reply:
good catch! After rewriting `zip` in terms of `zipWith`, there's no need to override `zip` here.